### PR TITLE
User add

### LINKED ISSUE #1197: Implement Account - User add / edit

### PR DESCRIPTION
**Pull request by na3d**
Closes #1197 
screen shot 2017-03-22 at 10 10 54 am

### CODE REVIEW

File: scss/components/buttons.scss [hunk 1]
```diff
@@ -79,6 +79,7 @@ a {
     &.btn-primary {
         &, &:active, &:focus {
             background: $blue;
+            color: white;
```
Comment (mparke) [inline scss/components/buttons.scss]: use var

File: src/users/components/UserForm.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            <FormGroupError errors={errors} name="username" />
+          </div>
+        </FormGroup>
+        <FormGroup errors={errors} name="email" className="row">
+          <label className="col-sm-2 col-form-label">Email</label>
+          <div className="col-sm-6">
+            <Input
+              id="user-email"
+              value={email}
+              onChange={e => this.setState({ email: e.target.value })}
```
Comment (mparke) [inline src/users/components/UserForm.js]: handle changes in a single function if possible: 

https://facebook.github.io/react/docs/forms.html#handling-multiple-inputs

File: src/users/components/UserForm.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      username,
+      email,
+      password,
+      restricted,
+      restrictedLabel,
+    } = this.state;
+
+    return (
+      <Form
+        onSubmit={async () => {
```
Comment (mparke) [inline src/users/components/UserForm.js]: handle submit in a class method like below for consistency?

File: src/users/layouts/AddUserPage.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    const { dispatch } = this.props;
+
+    this.setState({ loading: true });
+    try {
+      await dispatch(users.post(stateValues));
+      dispatch(push('/users'));
+    } catch (response) {
+      await new Promise(async (resolve) => this.setState({
+        loading: false,
+        errors: Object.freeze(await reduceErrors(response)),
```
Comment (mparke) [inline src/users/layouts/AddUserPage.js]: why the Object.freeze?

File: src/users/layouts/EditUserPage.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    const { initUsername } = this.state;
+
+    this.setState({ loading: true });
+    try {
+      await dispatch(users.put(stateValues, initUsername));
+      dispatch(push('/users'));
+    } catch (response) {
+      await new Promise(async (resolve) => this.setState({
+        loading: false,
+        errors: Object.freeze(await reduceErrors(response)),
```
Comment (eatonphil) [inline src/users/layouts/EditUserPage.js]: Weird seeing this frozen here...

File: src/users/layouts/AddUserPage.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  async onSubmit(stateValues) {
+    const { dispatch } = this.props;
+
+    this.setState({ loading: true });
+    try {
+      await dispatch(users.post(stateValues));
+      dispatch(push('/users'));
+    } catch (response) {
+      await new Promise(async (resolve) => this.setState({
```
Comment (mparke) [inline src/users/layouts/AddUserPage.js]: can you just call `this.setState`? why the promise and the async function

File: src/users/layouts/EditUserPage.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      }, resolve));
```
Comment (eatonphil) [inline src/users/layouts/EditUserPage.js]: Could we not create the Promise here?

```js
const errors = await reduceErrors(response);
this.setState({ errors, loading: false });
```

File: src/users/layouts/AddUserPage.js (hunk 4 above)
Comment (abemassry) [inline src/users/layouts/AddUserPage.js]: the precedent is from here https://github.com/linode/manager/blob/master/src/linodes/linode/settings/layouts/EditConfigPage.js#L305

File: src/users/layouts/EditUserPage.js (hunk 5 above)
Comment (abemassry) [inline src/users/layouts/EditUserPage.js]: the precedent is from here
https://github.com/linode/manager/blob/master/src/linodes/linode/settings/layouts/EditConfigPage.js#L305

File: src/users/components/UserForm.js (hunk 3 above)
Comment (na3d) [inline src/users/components/UserForm.js]: As long as we do it the same way for both PRs that are doing this.

File: src/users/layouts/EditUserPage.js (hunk 7 above)
Comment (na3d) [inline src/users/layouts/EditUserPage.js]: Should it not be used [here](https://github.com/linode/manager/blob/9b848c314ee428ddd71ced66dd9f67e8db7feff7/src/linodes/linode/settings/layouts/EditConfigPage.js#L303) as well?

File: src/users/layouts/AddUserPage.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+          <div className="PrimaryPage-headerRow clearfix">
+            <h1>Users</h1>
+          </div>
+        </header>
+        <div className="PrimaryPage-body User-create">
+          <Card title="Add a user">
+            <UserForm
+              restricted={false}
+              restrictedLabel="Restricted user
+                (Customize restriction)"
```
Comment (eatonphil) [inline src/users/layouts/AddUserPage.js]: These two phrases don't make very much sense together... What do you mean?

**Comment by eatonphil** [comment]
Is this missing screenshots of other pages?

**Comment by na3d** [comment]
screen shot 2017-03-22 at 4 35 01 pm

**Comment by na3d** [comment]
That is the only other page.

**Comment by na3d** [comment]
Oh yeah, forgot that buttons got added here:
screen shot 2017-03-22 at 4 40 48 pm

**Pull request merged**